### PR TITLE
adding user_undel and adding parameters to user_add

### PR DESCRIPTION
**Pull request by hzgraham**
In client.py I'm adding user_undel to allow restoring of preserved users.
I am also adding the gid_number and the uid_number parameters to user_add so that the uidNumber and gidNumber can be set upon user creation

### CODE REVIEW

**Comment by AmbientLighter** [comment]
@hzgraham These changes look sane to me. Could you please however fix linter issues so that I can merge this PR?

**Comment by hzgraham** [comment]
@AmbientLighter thanks, I pushed a fix for that issue. It seems to have passed, but let me know if there is anything else.

File: src/python_freeipa/client.py [hunk 1]
```diff
@@ -566,6 +576,17 @@ def user_del(self, username, skip_errors=False, soft_delete=False):
         }
         return self._request('user_del', username, params)
 
+    def user_undel(self, username):
+        """
+        Undelete a user.
+
+        :param username: User login.
+        :type username: string
+        :param skip_errors: Continuous mode: Don't stop on errors.
```
Comment (AmbientLighter) [inline src/python_freeipa/client.py]: Hm, this param is not defined in method signature. Could you please either remove this param from docstring or add it to the method signature?

**Comment by hzgraham** [comment]
@AmbientLighter I wasn't exactly sure what you wanted me to remove, but I removed the "skip_errors" parameter from the doc string. Let me know if there is anything else.

**Pull request merged**